dftimewolf/lib/collectors/aws.py

```diff
@@ -0,0 +1,252 @@
+      all_volumes (bool): Optional. True if all volumes attached to the source
+          instance should be copied.
+      analysis_profile_name (str): Optional. The AWS account in which to
+          create the analysis VM. This is the profile name that is defined in
+          your AWS credentials file.
+      analysis_zone (str): Optional. The AWS zone in which to create the VM.
+          If not specified, the VM will be created in the same zone where the
+          volume(s) exist(s).
+      boot_volume_size (int): Optional. The size (in GB) of the boot volume
+          for the analysis VM. Default is 50 GB.
+      cpu_cores (int): Optional. The number of CPU cores to use for the
+          analysis VM. Default is 16.
+      ami (str): Optional. The Amazon Machine Image ID to use to create the
+          analysis VM. Default is Ubuntu 18.04 TLS.
+    """
+
+    if not (remote_profile_name and remote_zone):
+      self.state.AddError('You must specify "remote_profile_name" and "zone" '
+                          'parameters', critical=True)
+
+    self.remote_profile_name = remote_profile_name
+    self.remote_zone = remote_zone
+    self.source_account = aws_account.AWSAccount(
+        self.remote_zone, aws_profile=self.remote_profile_name)
+
+    self.incident_id = incident_id
+    self.remote_instance_id = remote_instance_id
+
+    self.volume_ids = volume_ids.split(',') if volume_ids else []
+    self.all_volumes = all_volumes
+    self.analysis_zone = analysis_zone or remote_zone
+    self.analysis_profile_name = analysis_profile_name or remote_profile_name
+
+    # See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html
+    self.device_names = [chr(i) for i in range(ord('f'), ord('p') + 1)]
+
+    analysis_vm_name = 'aws-forensics-vm-{0:s}'.format(self.incident_id)
+    print('Your analysis VM will be: {0:s}'.format(analysis_vm_name))
+    self.state.StoreContainer(
+        containers.TicketAttribute(
+            name=self._ANALYSIS_VM_CONTAINER_ATTRIBUTE_NAME,
+            type_=self._ANALYSIS_VM_CONTAINER_ATTRIBUTE_TYPE,
+            value=analysis_vm_name))
+    self.analysis_vm, _ = aws_forensics.StartAnalysisVm(
+        analysis_vm_name,
+        self.analysis_zone,
+        boot_volume_size,
+        ami=ami,
+        cpu_cores=cpu_cores,
+        dst_profile=self.analysis_profile_name,
+    )
+
+  def _GetVolumesFromIds(self, volume_ids):
+    """Gets volumes from an account by volume IDs.
+
+    Args:
+      volume_ids (list[str]): List of volume ids to get from the account.
+
+    Returns:
+      list[AWSVolume]: List of AWSVolume objects to copy.
+    """
+    volumes = []
+    for volume_id in volume_ids:
+      try:
+        volumes.append(self.source_account.GetVolumeById(volume_id))
+      except RuntimeError:
+        self.state.AddError(
+            'Volume "{0:s}" was not found in AWS account {1:s}'.format(
+                volume_id, self.remote_profile_name),
+            critical=True)
```

Comment:
If the error is critical, then you will want to return an empty list, and check for the return value of this in the caller (I know, I should really fix #80)